motr/m0crate/crate_index.c

```diff
@@ -408,12 +481,15 @@ static int cr_idx_w_init(struct cr_idx_w *ciw,
 	else
 		ciw->op_selector = CR_OP_SEL_RR;
 
-	/* XXX: if DEL and PUT balanced, then we have to
+	/* if DEL or GET or NEXT and PUT given, then we have to
 	* use round-robin selector to avoid cases when
 	* we perform GET/DEL/NEXT on empty index.
 	*/
-	if (ciw->nr_ops[CRATE_OP_DEL].nr == ciw->nr_ops[CRATE_OP_PUT].nr)
+	if ((ciw->nr_ops[CRATE_OP_DEL].nr || ciw->nr_ops[CRATE_OP_GET].nr ||
```

Comment:
what is impact of removing equality check ? `==`

Reply:
As such there is no impact if PUT==DEL are equal, because both are non-zero, eventually condition would be true in that case. Here we want avoid key not exits issue in get, del and next ops.

Reply:
according to our /doc/coding-style.md
```
137   * use difference between NULL, 0 and "false" to emphasize whether
138     an expression is used as a pointer, integer (including success
139     or failure code) or boolean:
140 
141           if (p == NULL) { /* assumes that p is a pointer */
142           } else if (q != 0) { /* q is an integer */
143           } else if (r) { /* r is a boolean */
144           }

```
This should be checked as:
if ((ciw->nr_ops[CRATE_OP_DEL].nr != 0 || ciw->nr_ops[CRATE_OP_GET].nr != 0 || 
     ciw->nr_ops[CRATE_OP_NEXT].nr != 0) && ciw->nr_ops[CRATE_OP_PUT].nr != 0)

Reply:
Fixed.